src/test/java/ru/hogwarts/school/FacultyControllerTest.java

```diff
@@ -0,0 +1,102 @@
+package ru.hogwarts.school;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import ru.hogwarts.school.controllers.FacultyController;
+import ru.hogwarts.school.model.Faculty;
+
+import static org.assertj.core.api.AssertionsForClassTypes.*;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class FacultyControllerTest {
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private FacultyController facultyController;
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+    @Test
+    void contextLoads() throws Exception {
+        assertThat(facultyController).isNotNull();
+    }
+
+    @Test
+    public void getAllFacultiesTest() throws Exception {
+        assertThat(this.testRestTemplate.getForObject("http://localhost:" + port + "/faculty/all", String.class))
+                .isNotNull();
+    }
+
+    @Test
+    public void getFacultyByIdTest() throws Exception {
+        long id = 1;
+        assertThat(this.testRestTemplate.getForObject("http://localhost:" + port + "/faculty/" + id, String.class))
+                .isNotNull();
+    }
+
+    @Test
+    public void getFacultyByName() throws Exception {
+        String name = "aaa";
+        assertThat(this.testRestTemplate.getForObject("http://localhost:" + port + "/faculty/" + name, String.class))
+                .isNotNull();
+    }
+
+    @Test
+    public void createFacultyTest() throws Exception {
+        Faculty faculty = new Faculty();
+        faculty.setName("eee");
+        faculty.setColor("green");
+
+        assertThat(this.testRestTemplate.postForObject("http://localhost:" + port + "/faculty", faculty, String.class))
+                .isNotNull();
+    }
+
+    @Test
+    public void editFacultyTest() throws Exception {
+        Faculty faculty = new Faculty();
+        faculty.setName("ooo");
+        faculty.setColor("brown");
+
+        assertThat(this.testRestTemplate.postForObject(
+                "http://localhost:" + port + "/faculty/" + faculty.getId(), faculty, String.class))
+                .isNotNull();
+    }
+
+    @Test
+    public void deleteFacultyTest() throws Exception {
+        Faculty faculty = new Faculty();
+        faculty.setName("ppp");
+        faculty.setColor("white");
+        faculty.setId(10L);
+
+        assertThat(this.testRestTemplate.postForObject("http://localhost:" + port, faculty, String.class))
+                .isNotNull();
+        long id = faculty.getId();
```

Comment:
```suggestion
        Faculty actual = this.testRestTemplate.postForObject("http://localhost:" + port + "/faculty/", faculty, Faculty.class);
        assertThat(actual).isNotNull();
        long id = actual.getId();
```
Внеси такие правки. Вот почему:
- опечатка в url'e
- hibernate сгенерирует идентификатор независимо от того, какой ты передашь, поэтому нужно достать id новой записи